kdoctor/src/commonMain/kotlin/org/jetbrains/kotlin/doctor/diagnostics/GradleProjectDiagnostic.kt

```diff
@@ -0,0 +1,186 @@
+        var gradleVersion = system.execute(gradleBin, "-v")
+        if (gradleVersion.code != 0) {
+            gradleBin = "gradle"
+            gradleVersion = system.execute(gradleBin, "-v")
+            if (gradleVersion.code != 0) {
+                result.addFailure(
+                    "Gradle is not found",
+                    "For better experience we recommend to use a Gradle wrapper"
+                )
+                return result.build()
+            } else {
+                result.addInfo(
+                    "Project doesn't use a Gradle wrapper",
+                    "System Gradle was used for the analyse",
+                    "For better experience we recommend to use a Gradle wrapper"
+                )
+            }
+        }
+
+        val gradle = gradleVersion.output.orEmpty().lines()
+            .firstOrNull { it.startsWith("Gradle ") }
+            ?.let { Application("Gradle", Version(it.substringAfter("Gradle "))) }
+
+        if (gradle == null) {
+            result.addFailure(
+                "Gradle is not found",
+                "For better experience we recommend to use a Gradle wrapper"
+            )
+            return result.build()
+        } else {
+            result.addSuccess(
+                """${gradle.name} (${gradle.version})"""
+            )
+        }
+
+        val tempFile = try {
+            system.writeTempFile(printPluginsInitScript())
+        } catch (e: Exception) {
+            result.addFailure(
+                "Error: impossible to create temporary file",
+                "Check your file system write permissions",
+                "'${e.message}'"
+            )
+            return result.build()
+        }
+        val initScript = tempFile.substringBeforeLast("/") + "/init.gradle.kts"
+
+        system.execute("rm", initScript) //result code doesn't matter
+
+        val initScriptRenamed = system.execute("mv", tempFile, initScript).code
+        if (initScriptRenamed != 0) {
+            result.addFailure(
+                "Error $initScriptRenamed: impossible to prepare $initScript file",
+                "Check your file system write permissions"
+            )
+            return result.build()
+        }
+
+        Logger.d("Created $initScript")
+
+        val runGradle = system.execute(gradleBin, "-p", projectPath, "-I", initScript)
+
+        if (runGradle.code != 0) {
+            result.addFailure(
+                "Gradle error",
+                "Run in terminal '$gradleBin -p $projectPath --info'"
+            )
+            return result.build()
+        }
+
+        val gradleOutput = runGradle.output.orEmpty()
+        val pluginArtifacts = mutableListOf<PluginArtifact>()
+        val appliedPluginsMap = mutableMapOf<String, String>()
+        gradleOutput.lineSequence().forEach { l ->
+            if (l.contains(KDOCTOR_PREFIX)) {
+                val line = l.substringAfter(KDOCTOR_PREFIX)
+                if (line.startsWith("DEP ")) {
```

Comment:
Maybe it's worth extracting `DEP ` to a constant?

Reply:
I decided that this is a private internal key for one specific function.